src/utils/create-slots.tsx

```diff
@@ -82,7 +83,7 @@ const createSlots = <SlotNames extends string>(slotNames: SlotNames[]) => {
   }> = ({name, children}) => {
     const {registerSlot, unregisterSlot, context} = React.useContext(SlotsContext)
 
-    React.useEffect(() => {
+    useLayoutEffect(() => {
```

Comment:
I was running into really weird issues with the file upload where the editor content wouldn't get updated when the file upload completed. I traced it down to the commit where I introduced slots. I have to admit I don't full understand why this was happening or why this fix works, but I think it is because the file upload handler is asynchronous and so is `useEffect`; there must have been some clashing here where a render got 'lost'. Changing this to `useLayoutEffect` fixed the problem and slots still seem to work fine in every component that uses them.